### PR TITLE
Handle empty strings

### PR DESCRIPTION
**Pull request by landonreed**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This PR addresses ibi-group/datatools-ui#619 and ibi-group/datatools-ui#616 by better handling empty string values passed to JdbcTableWriter for updates to GTFS entities. It also enhances tests to verify that `trip_id` is auto-generated for empty string literals.

### CODE REVIEW

**Review by evansiroky — APPROVED** [review]

**Review by binh-dam-ibigroup — APPROVED** [review]
LGTM, see comments about setting the json field to null in another code block.

File: src/main/java/com/conveyal/gtfs/loader/JdbcTableWriter.java [hunk 1]
```diff
@@ -417,25 +418,33 @@ private void setStatementParameters(
...
+                            field.setParameter(preparedStatement, index, text);
+                        }
                     }
                 }
             } catch (StorageException e) {
                 LOG.warn("Could not set field {} to value {}. Attempting to parse integer seconds.", field.name, value);
                 if (field.name.contains("_time")) {
                     // FIXME: This is a hack to get arrival and departure time into the right format. Because the UI
-                    // currently returns them as seconds since midnight rather than the Field-defined format HH:MM:SS.
+                    //  currently returns them as seconds since midnight rather than the Field-defined format HH:MM:SS.
```
[the remark this answers is not in the document]
Comment (landonreed) [reply src/main/java/com/conveyal/gtfs/loader/JdbcTableWriter.java:447]: The extra space tells Intellij to format the text as a single FIXME block:

with extra space:
image

without:
image

File: src/main/java/com/conveyal/gtfs/loader/JdbcTableWriter.java [hunk 2]
```diff
@@ -417,25 +418,33 @@ private void setStatementParameters(
...
-                        // If the field is a ShortField and the string is empty, set value to null. Otherwise, set
-                        // parameter with string.
-                        if (field instanceof ShortField && text.isEmpty()) field.setNull(preparedStatement, index);
-                        else field.setParameter(preparedStatement, index, text);
+                        // If the string is empty, set value to null (for StringField, ShortField, etc.). Otherwise, set
+                        // parameter with string value.
+                        if (text.isEmpty()) {
+                            // Update JSON object that is returned in response (to reflect database value).
+                            // Note: see return value of JdbcTableWriter#update.
+                            jsonObject.set(field.name, null);
```
[the remark this answers is not in the document]
Comment (landonreed) [reply src/main/java/com/conveyal/gtfs/loader/JdbcTableWriter.java:436]: Good question. I'll do some testing with the UI to see how this interaction works.
